sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/HttpPipelineHelper.cs

```diff
@@ -126,5 +129,63 @@ internal static bool TryGetRetryIntervalTimespan(Response? httpResponse, out Tim
 
             return Encoding.UTF8.GetBytes(partialContent);
         }
+
+        internal static ExportResult IsSuccess(HttpMessage httpMessage)
+        {
+            if (httpMessage.HasResponse && httpMessage.Response.Status == ResponseStatusCodes.Success)
+            {
+                return ExportResult.Success;
+            }
+
+            return ExportResult.Failure;
+        }
+
+        internal static void HandleFailures(HttpMessage httpMessage, PersistentBlob blob, PersistentBlobProvider blobProvider)
+        {
+            int statusCode = 0;
+            bool shouldRetry = true;
+
+            if (httpMessage.HasResponse)
+            {
+                statusCode = httpMessage.Response.Status;
+                switch (statusCode)
+                {
+                    case ResponseStatusCodes.PartialSuccess:
+                        // Parse retry-after header
+                        // Send Failed Messages To Storage
```

Comment:
This will create a new timestamp on the file name. Instead should be just release the lease on the blob?

Reply:
This is a special case. A new blob will be generated due to partial transmission. (see line 159-163)